### PR TITLE
 Convert views/private Slim templates to ERB

### PR DESCRIPTION
**Pull request by spavlishak**

### CODE REVIEW

File: app/views/private/api_tokens/_user_tokens.html.erb [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+                <tr>
+                  <td class="token-label">
+                    <%= token.label %>
+                  </td>
+                  <td class="access_key">
+                    <%= token.access_key %>
+                  </td>
+                  <td class="actions">
+                    <%= link_to t('.edit'), edit_api_token_path(token) %>
+                    | &nbsp;
```
Comment (yivo) [inline app/views/private/api_tokens/_user_tokens.html.erb]: @spavlishak check text content (remove pipe)

File: app/views/private/assets/_btc_assets.html.erb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        <h3><%= t('.title') %></h3>
+        <p class="text-ignore"><%= t('.intro') %></p>
+      </div>
+      <div id="content" class="col-xs-16">
+        <div class="jumbotron">
+          <div class="row">
+            <div class="col-xs-offset-4 col-xs-16">
+              <span class="pull-left"><%= t('.total') %></span>
+              <span class="pull-right">
+                <%= t('currency.icon.btc') %>
```
Comment (yivo) [inline app/views/private/assets/_btc_assets.html.erb]: @spavlishak this should not escaped

File: app/views/private/assets/_cny_assets.html.erb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        <h3><%= t('.title') %></h3>
+        <p class="text-ignore"><%= t('.intro') %></p>
+      </div>
+      <div id="content" class="col-xs-16">
+        <div class="jumbotron">
+          <div class="row">
+            <div class="col-xs-offset-4 col-xs-16">
+              <span class="pull-left"><%= t('.total') %></span>
+              <span class="pull-right">
+                <%= t('currency.icon.cny') %>
```
Comment (yivo) [inline app/views/private/assets/_cny_assets.html.erb]: @spavlishak this should not be escaped

File: app/views/private/assets/_cny_assets.html.erb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+                <%= @cny_proof.balance %>
+              </span>
+            </div>
+          </div>
+          <div class="clearfix"></div>
+        </div>
+
+        <%= t('.description', @cny_assets['accounts'].first.symbolize_keys) %>
```
Comment (yivo) [inline app/views/private/assets/_cny_assets.html.erb]: @spavlishak this should not be escaped

File: app/views/private/assets/_liability_data.html.erb [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<div class="proof">
+  <% partial_tree = proof.try(:partial_tree_of, account) %>
+  <% if proof.try(:ready?) && partial_tree.present? %>
+    <% root_target = "#{proof.currency}_root_json" %>
+    <% partial_tree_target = "#{proof.currency}_partial_tree_json" %>
+
+    <div class="root json">
+      <h5><%= t('.root-json') %></h5>
+      <pre id="root_target"><%= proof.root.to_json %></pre>
```
Comment (yivo) [inline app/views/private/assets/_liability_data.html.erb]: @spavlishak you should set ID to variable, not fixed value

File: app/views/private/assets/_liability_data.html.erb [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      <div class="clearfix"></div>
+    </div>
+
+    <div class="partial-tree json">
+      <div class="assets">
+        <span class="title">
+          <%= t('.partial-tree', currency: proof.currency.upcase) %>
+        </span>
+        <span class="currency">
+          <%= t("currency.icon.#{proof.currency}") %>
```
Comment (yivo) [inline app/views/private/assets/_liability_data.html.erb]: @spavlishak this should not be escaped

File: app/views/private/assets/_liability_data.html.erb [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+          <%= partial_tree.sum %>
+        </span>
+      </div>
+      <pre id="partial_tree_target"><%= partial_tree.json.to_json %></pre>
```
Comment (yivo) [inline app/views/private/assets/_liability_data.html.erb]: @spavlishak you should assign var here

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by yivo** [comment]
This is not a complete review. I will review again when current issues will be resolved.

**Comment by yivo** [comment]
@spavlishak Please rebase.

File: app/views/private/assets/_liability_data.html.erb [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      </div>
+      <pre id="<%= partial_tree_target %>"><%= partial_tree.json.to_json %></pre>
+      <div class="clearfix"></div>
+    </div>
+
+    <div class="timestamp">
+      <%= t('.timestamp', time: proof.timestamp) %>
+    </div>
+
+    <hr></hr>
```
Comment (yivo) [inline app/views/private/assets/_liability_data.html.erb]: @spavlishak `hr` is self-closing tag.

File: app/views/private/assets/_xrp_assets.html.erb [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+      </div>
+      <div id="content" class="col-xs-16">
+        <div class="jumbotron">
+          <div class="row">
+            <div class="col-xs-offset-4 col-xs-16">
+              <span class="pull-left">
+                <%= t('.total') %>
+              </span>
+              <span class="pull-right">
+                <%= t('currency.icon.xrp') %>
```
Comment (yivo) [inline app/views/private/assets/_xrp_assets.html.erb]: @spavlishak this should not be escaped.

**Comment by spavlishak** [comment]
@yivo rebased

**Review by yivo — CHANGES_REQUESTED** [review]

**Comment by yivo** [comment]
Rebased.

File: app/views/private/api_tokens/_oauth_tokens.html.erb [hunk 10]
```diff
@@ -0,0 +1,63 @@
+<div class="panel panel-default">
```
Comment (gfedorenko) [inline app/views/private/api_tokens/_oauth_tokens.html.erb]: _oauth_tokens.html.slim was removed so _oauth_tokens.html.erb should be removed too

File: app/views/private/markets/_account_summary.html.erb [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  </a>
+  <div class="dropdown-menu text-right">
+    <p class="text-right">
+      <ul class="list-inline text-right"></ul>
+    </p>
+    <table class="table">
+      <tbody>
+        <% current_user.accounts.each do |account|
+          next unless account.currency_obj.try(:visible) %>
+          <tr class="<% account.currency %> <%= @market.scope?(account) ? '' : 'hide' %>">
```
Comment (gfedorenko) [inline app/views/private/markets/_account_summary.html.erb]: I guess it should be <%= account.currency %> as below

File: app/views/private/markets/_market_list.html.erb [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+
+
+
+
+
+
+
+
+<div class="has-right-dropdown" id="market_list_wrapper">
```
Comment (gfedorenko) [inline app/views/private/markets/_market_list.html.erb]: Do we really need 62-107 lines?

File: app/views/private/shared/_balances.html.erb [hunk 13]
```diff
@@ -0,0 +1,17 @@
+<div class="row">
+  <% member.accounts.each do |account| %>
+    <div class="col-xs-6">
+      <div class="row <%= account - account.currency %>">
```
Comment (gfedorenko) [inline app/views/private/shared/_balances.html.erb]: Should be account-<%= account.currency %>

File: app/views/private/withdraws/index.html.erb [hunk 14]
```diff
@@ -0,0 +1,36 @@
+<ul class="list-group">
+  <% @channels.each do |c| %>
+    <li class="list-group-item withdraws-channel-<% c.id %>">
```
Comment (gfedorenko) [inline app/views/private/withdraws/index.html.erb]: Should be <%= c.id %>

**Comment by yivo** [comment]
Rebased once again.

**Comment by yivo** [comment]
@spavlishak Please, check your code and fix typos where you omit `=` in ERB output.

**Comment by yivo** [comment]
Rebased.

File: app/views/private/assets/index.html.erb [hunk 15]
```diff
@@ -0,0 +1,27 @@
+<%= content_for :frame do %>
+  <div class="container">
+    <ul class="nav nav-justified assets-nav">
+      <li class="active" data-scroll-nav='0'>
```
Comment (gfedorenko) [inline app/views/private/assets/index.html.erb]: Should be data-scroll-nav="0"

File: app/views/private/assets/index.html.erb [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+        <a><%= t('.liability-proof') %></a>
+      </li>
+      <li data-scroll-nav='1'>
+        <a><%= t('.btc-assets') %></a>
+      </li>
+      <li data-scroll-nav='2'>
+        <a><%= t('.xrp-assets') %></a>
+      </li>
+      <li data-scroll-nav='3'>
```
Comment (gfedorenko) [inline app/views/private/assets/index.html.erb]: Same for all data-scroll-nav props

File: app/views/private/markets/_market_list.html.erb [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+              </th>
+            </tr>
+          </thead>
+        </table>
+      </div>
+      <div class="panel-body panel-body-content">
+        <table class="table table-hover markets all">
+          <tbody>
+            <% @markets.each do |market| %>
+              <tr class="market" id="market-list-<% market.id %>" class="quote-<% market.quote_unit%>" data-market="<% market.id %>">
```
Comment (gfedorenko) [inline app/views/private/markets/_market_list.html.erb]: Should be <%= market.id %> <%= market.quote_unit%> <%= market.id %>

File: app/views/private/markets/_range_switch.html.erb [hunk 18]
```diff
@@ -0,0 +1,13 @@
+<ul id="range_switch" class="list-inline switch" data-x="60">
+  <% %w(1 5 15 30 | 60 120 240 360 720 | 1440 4320 | 10080).each do |x| %>
+    <li class="text-center">
+      <% if x == '|' %>
+        <span> | </span>
+      <% else %>
+        <a class="hand-point chart-switch" href="#" data-x="<% x %>">
```
Comment (gfedorenko) [inline app/views/private/markets/_range_switch.html.erb]: <%= x %>

File: app/views/private/markets/_market_list.html.erb [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+    <div class="dropdown-wrapper">
+      <ul class="dropdown-menu">
+        <li>
+          <a class="active" data-name="all" href="javascript:;">
+            <%= t('.all_html') %>
+          </a>
+        </li>
+        <% @market_groups.each do |name| %>
+          <li>
+            <a data-name="<% name %>" href="javascript:;">
```
Comment (gfedorenko) [inline app/views/private/markets/_market_list.html.erb]: <%= name %>

File: app/views/private/deposits/ripples/new.html.erb [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+<%= content_for :guide %>
+  <%= image_tag('coins/xrp') %>
+<% end %>
+
+<%= content_for :after %>
+  <%= panel do %>
+    <%= table_for(@assets, class: 'table table-condensed table-hover', model: @model) do |t| %>
+      <% t.column :created_at, class: 'col-xs-3' %>
+      <% t.column :txid, class: 'col-xs-4' do |x| %>
+        <a href="<% x.blockchain_url %>" target="_blank">
```
Comment (gfedorenko) [inline app/views/private/deposits/ripples/new.html.erb]: <%= x.blockchain_url %>

File: app/views/private/assets/_liability_tabs.html.erb [hunk 21]
```diff
@@ -0,0 +1,57 @@
+<div class="tabbable">
+  <ul class="nav nav-tabs">
+    <li class="active">
+      <a class="btc-proof" href="#btc-proof" data-toggle="tab"><%= t('.verify-btc') %></a>
```
Comment (gfedorenko) [inline app/views/private/assets/_liability_tabs.html.erb]: Check this file for missing html_safe

File: app/views/private/assets/_usd_assets.html.erb [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+      </div>
+      <div id="content" class="col-xs-16">
+        <div class="jumbotron">
+          <div class="row">
+            <div class="col-xs-offset-4 col-xs-16">
+              <span class="pull-left">
+                <%= t('.total') %>
+              </span>
+              <span class="pull-right">
+                <%= t('currency.icon.usd') %>
```
Comment (gfedorenko) [inline app/views/private/assets/_usd_assets.html.erb]: Check for html_safe

File: app/views/private/markets/_header.html.erb [hunk 23]
```diff
@@ -0,0 +1,36 @@
...
+      </p>
+    </ul>
+
+    <ul class="nav navbar-nav navbar-right">
+      <% if current_user %>
+        <%= render partial: 'account_summary' %>
+        <%= render partial: 'settings' %>
+      <% else %>
+        <p class="navbar-text">
+          <%= t('private.markets.mask.warning') %>
```
Comment (gfedorenko) [inline app/views/private/markets/_header.html.erb]: html_safe

File: app/views/private/markets/_mask.html.erb [hunk 24]
```diff
@@ -0,0 +1,5 @@
+<% unless current_user %>
+  <div class="mask">
+    <p class="text-center"><%= "#{t('.warning')}" %></p>
```
Comment (gfedorenko) [inline app/views/private/markets/_mask.html.erb]: html_safe

Comment (yivo) [reply app/views/private/markets/_mask.html.erb]: Also string interpolation is not necessary here.

**Comment by yivo** [comment]
@spavlishak Some issues: DON'T TOUCH CSS, ISSUE IS IN HTML

screen shot 2018-01-09 at 4 17 44 pm

**Comment by yivo** [comment]
screen shot 2018-01-09 at 4 26 58 pm

But should be

screen shot 2018-01-09 at 4 26 05 pm

**Comment by yivo** [comment]
Missing HTML safe at http://localhost:3000/exchange_assets (don't forget to exec rake task):

screen shot 2018-01-09 at 4 30 28 pm

**Review by yivo — APPROVED** [review]

**Comment by yivo** [comment]
@gfedorenko Could you please confirm that three issues above have been fixes?

Also please validate:
- [ ] There were no any HTML replacements for migration from Bootstrap 3 to 4.
- [ ] Confirm there are no visual glitches.
- [ ] Check `html_safe` issues & ERB output issues.

**Comment by gfedorenko** [comment]
Missing active class on the buttons on history page 

screenshot from 2018-01-10 17-35-23

screenshot from 2018-01-10 17-35-46

**Comment by gfedorenko** [comment]
Missing top panel on the trade page
screenshot from 2018-01-10 17-38-40
screenshot from 2018-01-10 17-38-51

**Pull request merged**